docs/authentication.md

```diff
@@ -0,0 +1,109 @@
+# The secure-proxy authentication flow
+
+There are 3 main actors to allow the user to authenticate themselves and to use
+the proxy:
+
+- the extension (SP)
+- secure-proxy REST API (SPS)
+- FxA services
+
+The steps are described here:
+https://gitlab.com/shane-tomlinson/mermaid-charts/blob/master/charts/secure-proxy/secure-proxy-signin-with-backend-server.svg
+
+To summarize them, the authentication flow works in this way:
+
+1. The SP sends a "state token request".
+2. The SP starts the authentication, calling the FxA authorization endpoint, and passing the state token as parameter.
+3. The SP obtains the fxa code from the FxA authorization endpoint.
+4. The SP sends a "finalize the authentication request" to SPS.
+
+To obtain the token, SP sends an "Info request" to know if the user has
+available passes/tokens. If there are passes available, the SP sends a "New
+token request".
+
+## SPS REST API
+
+### state token request
+This request returns a state token. The token is part of the response JSON
+object and it's sent as a cookie.
```

Comment:
Can we say "The token is sent in the JSON response and set in a session variable" ?